### ISSUE TITLE
Unhandled JS Exception when upgrading core to latest (5.15.2)

### ISSUE THREAD

**Issue by lukewlms** [react-navigation/react-navigation#9349]
**Current Behavior**

I'm trying to upgrade my react-navigation dependencies a few minor versions.

I can upgrade `/stack` just fine, but when I upgrade `/native` I'm getting the error below.  The error persists if I revert Native to the last working version, which makes me think it's in Core and not in Native.  Reverting those changes consistently fixes it.  Any clues here?

<img width="554" alt="image" src="https://user-images.githubusercontent.com/19214773/108702801-1890b380-754d-11eb-9d84-eb1b4589ed14.png">

**Expected Behavior**

No error

**How to reproduce**

Unfortunately I can't [specify a resolution](https://github.com/yarnpkg/yarn/issues/4874) for `/core` and `/routers` so I'm not sure how to narrow it down other than that old versions (core@5.12.3, routers@5.4.11) work fine but once Native is upgraded and newest versions are fetched (core@5.15.2, routers@5.7.2) this error is constantly reproing.

I'm thinking since there are very-new versions out this may be affecting more people - logging for potential quick resolution, if more repro details are needed I could work on it.

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | 14.4 (iOS simulator)
| @react-navigation/native       | 5.7.3
| react-native                   | 0.63.4
| expo                           | N/A
| node                           | 12.18.3
| npm or yarn                    | 1.22.4

**Comment by lukewlms** [comment]
@github-actions yes.

**Comment by CptFabulouso** [comment]
Same here, `"@react-navigation/native": "5.7.6"` was working, after updating to currently latest `5.9.3` the error above appeared.

**Comment by raajnadar** [comment]
Try clearing the metro cache `npx react-native start --reset-cache` and also make sure you run `npx pod-install`

**Comment by CptFabulouso** [comment]
Yeah I did all that basic stuff. It works in another project, so it will be project related, I'll try to find what's causing it.

**Comment by CptFabulouso** [comment]
I had to do what you wrote here https://github.com/react-navigation/react-navigation/issues/7981#issuecomment-611377210 what was missing was deleting `yarn.lock`.

After running `yarn why @babel/core` I noticed there were references to versions like 7.2.x no matter what was in my `package.json`, after deleting `yarn.lock` the "correct" versions were installed. I installed the latest versions (7.13.10) for `@babel/core` and `@babel/runtime` and everything seems to be running fine again.
But I am not sure if deleting `yarn.lock` is good and safe practice, probably not in teams and maybe there is better way? Also not sure if it's safe to be updating babel to latest, instead of what the react sets up during init.
Bottom line is that it's most probably caused by babel.

**Comment by raajnadar** [comment]
Closing this issue

**Comment by lukewlms** [comment]
@CptFabulouso Thanks, I was also able to fix this by blowing away yarn.lock and regenerating.
